### ISSUE TITLE
Problem during isoformSwitchAnalysisPart 1 at step 3 of 3

### ISSUE THREAD

**Issue by Pinolinoo** [kvittingseerup/IsoformSwitchAnalyzeR#122]
hi kristoffer, 

I am getting an error during step 3 of  isoformSwitchAnalysisPart 1() :

'Step 1 of 3 : Detecting isoform switches... Step 2 of 3 : Adding and predicting open reading frames Step 3 of 3 : Extracting (and outputting) sequences Fehler in extractSequence(switchAnalyzeRlist = switchAnalyzeRlist, genomeObject = genomeObject, : The object supplied to 'switchAnalyzeRlist' must be a 'switchAnalyzeRlist' Zusätzlich: Warnmeldungen: 1: In DESeqDataSet(rse, design, ignoreRank = TRUE) : some variables in design formula are characters, converting to factors 2: In addORFfromGTF(switchAnalyzeRlist = switchAnalyzeRlist, pathToGTF = pathToGTF, : It seems ORF was only added to an unlikely small fraction of isoforms (less than 10%).If you are not sure this is on purpose something went wrong and the files are not matching.'

`> aSwitchlist <- preFilter(aSwitchList)

The filtering removed 42174 ( 38.25% of ) transcripts. There is now 68090 isoforms left
`

`> aSwitchlist <- isoformSwitchTestDEXSeq(aSwitchlist)

Step 1 of 2: Testing each pairwise comparisons with DEXSeq (this might be a bit slow)...

    Estimated run time is: 56.3 min

Step 2 of 2: Integrating result into switchAnalyzeRlist...

    Isoform switch analysis was performed for 9440 gene comparisons (100%).

Total runtime: 21.55 min

Done

Warnmeldung:

In DESeqDataSet(rse, design, ignoreRank = TRUE) :
  some variables in design formula are characters, converting to factors`

`> aSwitchlist <- addORFfromGTF(aSwitchlist, pathToGTF = "~/Desktop/phd/isoformswitch /hg38.gtf")

Step 1 of 2: importing GTF (this may take a while)...

Step 2 of 2: Adding ORF...

    Added ORF info (incl info about isoforms annotated as not having an ORF) to 11 isoforms.

        This correspond to 2.36% of isoforms in the switchAnalyzeRlist.

            Which includes 8.21% of isoforms from annotated genes (novel isoforms not counted) in the switchAnalyzeRlist.

    Isoforms with no ORF annoation are either due to incompatible annotation versions or novel isoforms.

        We estimate 8.21% of isoforms not annotated with ORF are from novel genes.

        Examples of isoforms where ORF annoation is still missing are:
            MSTRG.90837.12, MSTRG.15037.22, MSTRG.8094.12

Done.

Warnmeldung:

In addORFfromGTF(aSwitchlist, pathToGTF = "~/Desktop/phd/isoformswitch /hg38.gtf") :

  It seems ORF was only added to an unlikely small fraction of isoforms (less than 10%).If you are not sure this is on purpose something went wrong and the files are not matching.
`

Now an error occurs: (it says the argument is missing (without standart value)
`> aSwitchlist <- analyzeNovelIsoformORF(aSwitchlist)
Fehler in analyzeNovelIsoformORF(aSwitchlist) : 
  Argument "analysisAllIsoformsWithoutORF" fehlt (ohne Standardwert)`

**Comment by kvittingseerup** [comment]
Yes. You can look at the documentation of `analyzeNovelIsoformORF()` for more information about that.

Also - what do you think about the warnings from `addORFfromGTF()`?

Could you also post the result of running `summary(aSwitchlist)`

**Comment by Pinolinoo** [comment]
thanks! 

actually I don't really understand what open reading frames are!
sorry, still pretty new in the game :) 

 `This switchAnalyzeRlist list contains:
 466 isoforms from 60 genes
 1 comparison from 2 conditions (in total 18 samples)

Switching features:
      Comparison Isoforms Switches Genes
1 Healthy vs Ill       64       89    60

Feature analyzed:
[1] "Isoform Switch Identification, ORFs"`

**Comment by kvittingseerup** [comment]
I think you need to do some reading and sanity checks before proceeding. The numbers reported from `addORFfromGTF()` seems very low...

**Comment by Pinolinoo** [comment]
ok I will try to understand it better. 

So I get the sense there's an issue with my data and no bug in the code? This is why I get the error in analyzeNovelIsoformORF() ? I thought for you it ran with the exact same data?

**Comment by kvittingseerup** [comment]
The code runs (and finishes) so I doubt that it is the origin of the problem you get warnings about. If it turn out that it is please don't hesitate to let me know :-)
